### PR TITLE
API Update translation to use symfony 6

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10389

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
         "symfony/cache": "^6.1",
         "symfony/config": "^6.1",
         "symfony/filesystem": "^6.1",
-        "symfony/translation": "^4.4.44",
+        "symfony/translation": "^6.1",
+        "symfony/translation-contracts": "^3.1",
```
Comment (GuySartorelli) [inline composer.json]: Why is this new dependency needed?

**Comment by GuySartorelli** [comment]
@emteknetnz This needs to be rebased after the changes to the upstream PR

**Comment by emteknetnz** [comment]
Have rebased

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
This PR has two commits - the first one should be removed, it's a holdover from the old parent PR.

**Comment by emteknetnz** [comment]
Updated

**Review by GuySartorelli — APPROVED** [review]

**Pull request merged**